tests/profiles/googlefonts_test.py

```diff
@@ -2506,52 +2508,61 @@ def test_check_name_typographicfamilyname():
+  # Add an incorrect TYPOGRAPHIC_SUBFAMILY_NAME to a RIBBI font
+  ttFont['name'].setName("Generic subfamily name",
+                         NameID.TYPOGRAPHIC_SUBFAMILY_NAME,
+                         PlatformID.WINDOWS,
+                         WindowsEncodingID.UNICODE_BMP,
+                         WIN_ENGLISH_LANG_ID)
+  ttFont['name'].setName("Generic subfamily name",
+                         NameID.TYPOGRAPHIC_SUBFAMILY_NAME,
+                         PlatformID.MACINTOSH,
+                         UnicodeEncodingID.UNICODE_1_0,
+                         MAC_ROMAN_LANG_ID)
   print (f"Test FAIL with a RIBBI that has got a nameid={NameID.TYPOGRAPHIC_SUBFAMILY_NAME} entry...")
   status, message = list(check(ttFont,
-                               style_with_spaces(font)))[-1]
-  assert status == FAIL and message.code == "ribbi"
+                               expected_style(ttFont)))[-1]
+  assert status == FAIL
 
   # non-RIBBI fonts must have a TYPOGRAPHIC_SUBFAMILY_NAME entry
   font = TEST_FILE("montserrat/Montserrat-ExtraLight.ttf")
   ttFont = TTFont(font)
   print (f"Test PASS with a non-RIBBI containing a nameid={NameID.TYPOGRAPHIC_SUBFAMILY_NAME} entry...")
   status, message = list(check(ttFont,
-                               style_with_spaces(font)))[-1]
+                               expected_style(ttFont)))[-1]
   assert status == PASS
 
   # set bad values on all TYPOGRAPHIC_SUBFAMILY_NAME entries:
-  for i, name in enumerate(ttFont['name'].names):
-    if name.nameID == NameID.TYPOGRAPHIC_SUBFAMILY_NAME:
-      ttFont['name'].names[i].string = "foo".encode(name.getEncoding())
-
+  ttFont['name'].setName("Generic subfamily name",
+                         NameID.TYPOGRAPHIC_SUBFAMILY_NAME,
+                         PlatformID.WINDOWS,
+                         WindowsEncodingID.UNICODE_BMP,
+                         WIN_ENGLISH_LANG_ID)
+  ttFont['name'].setName("Generic subfamily name",
+                         NameID.TYPOGRAPHIC_SUBFAMILY_NAME,
+                         PlatformID.MACINTOSH,
+                         UnicodeEncodingID.UNICODE_1_0,
+                         MAC_ROMAN_LANG_ID)
   print (f"Test FAIL with a non-RIBBI with bad nameid={NameID.TYPOGRAPHIC_SUBFAMILY_NAME} entries...")
   status, message = list(check(ttFont,
-                               style_with_spaces(font)))[-1]
-  assert status == FAIL and message.code == "non-ribbi-bad-value"
+                               expected_style(ttFont)))[-1]
+  assert status == FAIL
 
   # remove all TYPOGRAPHIC_SUBFAMILY_NAME entries
-  # by changing their nameid to something else:
-  for i, name in enumerate(ttFont['name'].names):
-    if name.nameID == NameID.TYPOGRAPHIC_SUBFAMILY_NAME:
-      ttFont['name'].names[i].nameID = 255 # blah! :-)
-
+  win_name = ttFont['name'].getName(17, 3, 1, 1033)
+  mac_name = ttFont['name'].getName(17, 1, 0, 0)
+  win_name.nameID = 254
+  mac_name.nameID = 255
```

Comment:
yeah... in all these lines. Let's get rid of these hard-coded magic numbers that we, font-nerds, know by heart.

Reply:
Our nameIDs go up to 25. I think 254 and 255 are ok and we've used them in the past, https://github.com/googlefonts/fontbakery/blob/master/tests/profiles/googlefonts_test.py#L2497

Reply:
sure. I was referring to the `17, 3, 1, 1033` parts